control-plane-operator/controllers/hostedcontrolplane/v2/cloud_controller_manager/azure/deployment.go

```diff
@@ -33,5 +34,11 @@ func adaptDeployment(cpContext component.WorkloadContext, deployment *appsv1.Dep
 			azureutil.CreateVolumeForAzureSecretStoreProviderClass(config.ManagedAzureCloudProviderSecretStoreVolumeName, config.ManagedAzureCloudProviderSecretProviderClassName),
 		)
 	}
+
+	if azureutil.IsSelfManagedAzure() {
+		deployment.Spec.Template.Spec.ServiceAccountName = "azure-cloud-controller-manager"
+		deployment.Spec.Template.Spec.AutomountServiceAccountToken = ptr.To(true)
```

Comment:
this will be overridden here https://github.com/openshift/hypershift/blob/main/support/controlplane-component/defaults.go#L126-L131
why do we need this to be true?

Reply:
We may not need to. Let me test it without that. That may have just been leftover code from trying to get CCM working on self-managed Azure.